files/build_templates/sonic_debian_extension.j2

```diff
@@ -223,6 +223,8 @@ sudo cp {{src}} $FILESYSTEM_ROOT/{{dst}}
 
 {% if sonic_asic_platform == "mellanox" %}
 sudo mkdir -p $FILESYSTEM_ROOT/etc/mlnx/
-sudo cp target/files/$MLNX_FW $FILESYSTEM_ROOT/etc/mlnx/fw-SPC.mfa
-sudo cp platform/mellanox/mlnx-fw-upgrade.sh $FILESYSTEM_ROOT/usr/bin/
+sudo cp target/files/$MLNX_FW_FILE $FILESYSTEM_ROOT/etc/mlnx/fw-SPC.mfa
+j2 platform/mellanox/mlnx-fw-upgrade.j2 > platform/mellanox/mlnx-fw-upgrade.sh
```

Comment:
Linux will not allow it. To write something to image rootfs root permissions is required. There are two different operations: generation template with j2 and write it to file. With sudo will be executed only first one:
```
admin@arc-switch1026:~$ echo "OK" > /usr/bin/some_file
-bash: /usr/bin/some_file: Permission denied
admin@arc-switch1026:~$ sudo echo "OK" > /usr/bin/some_file
-bash: /usr/bin/some_file: Permission denied
admin@arc-switch1026:~$
```